packages/app/src/update/listeners/device-attached.listener.ts

```diff
@@ -5,13 +5,21 @@
 
 import { ListenerEvent } from "App/device-manager/constants"
 import { handleDeviceAttached } from "App/update/actions"
-import store from "App/__deprecated__/renderer/store"
+import store, { ReduxRootState } from "App/__deprecated__/renderer/store"
 import { ipcRenderer } from "electron-better-ipc"
+import { registerCurrentDeviceChangedListener } from "App/device-manager/listeners"
 
 // AUTO DISABLED - fix me if you like :)
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 const deviceAttachedHandler = (_: any, _data: string): void => {
   void store.dispatch(handleDeviceAttached())
+  if (
+    !ipcRenderer.listeners(ListenerEvent.CurrentDeviceChanged).length &&
+    !(store.getState() as unknown as ReduxRootState).dataSync
+      .initializationFailed
+  ) {
```

Comment:
I'm not sure about this condition. 
Seems hard to read. Check this:

> const noCurrentDeviceListeners =
>     ipcRenderer.listeners(ListenerEvent.CurrentDeviceChanged).length === 0
>   const { initializationFailed } = (
>     store.getState() as unknown as ReduxRootState
>   ).dataSync
> 
>   if (noCurrentDeviceListeners && !initializationFailed) {
>     registerCurrentDeviceChangedListener()
>   }

  
  I might be a little bit nitpicky... sorry :D
  @dkarski @lkowalczyk87 what you think?

Reply:
Well this is basically an answer to the flow we want to have.
If device takes to long to connect -> display modal -> closing modal should force the app to stop connecting to this device.
To get this kind of result i had to mound and unmount an listener at the start and end of this flow.
The first part of this condition checks if the listener is not mounted on the event
The second part is tricky because we use both rematch and redux toolkit to configure our store. This is the reason for all this type casting.
I don't really know how to tackle this issue in a different way :(